### PR TITLE
Add boilerplate remotewrite controller

### PR DESCRIPTION
**Pull request by mcbenjemaa**
A new controller needed to reconcile `RemoteWrite` CustomResources 
This is just the boilerplate and the actual implementation.
 
issue: https://github.com/giantswarm/roadmap/issues/1110

## Checklist

I have:

- [X] Described why this change is being introduced
- [ ] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/controller/remotewrite/controller.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/giantswarm/operatorkit/v7/pkg/controller"
+	promclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	vpa_clientset "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/client/clientset/versioned"
+
+	"github.com/giantswarm/prometheus-meta-operator/pkg/project"
+)
+
+type ControllerConfig struct {
+	K8sClient        k8sclient.Interface
+	Logger           micrologger.Logger
```
Comment (QuentinBisson) [inline service/controller/remotewrite/controller.go:16]: Can you skim this down to just the 3 first ? You will not use most of the rest if any

File: service/service.go [hunk 2]
```diff
@@ -272,6 +274,54 @@ func New(config Config) (*Service, error) {
 		}
 	}
 
+	var remotewriteController *remotewrite.Controller
+	{
+		if shouldCreateCAPIController(provider) {
```
Comment (QuentinBisson) [inline service/service.go:279]: Why is this only for CAPI Managemement cluster ? I do not think this filter is needed

File: service/controller/remotewrite/controller.go (hunk 1 above)
Comment (mcbenjemaa) [reply service/controller/remotewrite/controller.go:15]: Sure! I wasn't sure about those attributes.

File: service/service.go (hunk 2 above)
Comment (mcbenjemaa) [reply service/service.go]: Copy-paste stuff 😳

File: service/controller/remotewrite/controller.go (hunk 1 above)
Comment (QuentinBisson) [reply service/controller/remotewrite/controller.go:15]: Thanks :)

**Review by QuentinBisson — APPROVED** [review]

File: service/service.go [hunk 3]
```diff
@@ -272,6 +274,21 @@ func New(config Config) (*Service, error) {
...
+			Logger:           config.Logger,
+			PrometheusClient: prometheusClient,
+		}
+
+		remotewriteController, err = remotewrite.NewController(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+	}
```
Comment (TheoBrigitte) [inline service/service.go:290]: I would rather leave this out for now, and add it at last when we have resources and when everything works. So we keep PMO available for any upcoming and unforeseen fixes needed.

Also we this you need to boot the controller, basically adding a line further down in this file in the `Boot` method.

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]
Looks good, but would leave the final wiring out for now. Keep it in a separate PR so we can use it for testing.

File: service/service.go (hunk 3 above)
Comment (mcbenjemaa) [reply service/service.go]: Agree, `remotewriteController` would wired in a separate PR.

**Comment by TheoBrigitte** [comment]
CI is failing because of new nancy update which detect more vulnerabilities know. @hervenicol is working on a fix.

**Pull request merged**